react/Modal/index.jsx

```diff
@@ -3,57 +3,98 @@ import classNames from 'classnames'
+  const displayPrimary = primaryText && primaryAction
+  const displaySecondary = secondaryText && secondaryAction
+  return (displaySecondary || displayPrimary) &&
+    (
+      <div className={classNames(styles['coz-modal-content'], styles['coz-modal-buttons'])}>
+        { displaySecondary &&
+          <button className={classNames('coz-btn', 'coz-btn--' + secondaryType)} onClick={secondaryAction}>
+            {secondaryText}
+          </button>
         }
-        { children }
-        { ((cancelText && cancelAction) || (validateText && validateAction)) &&
-          <div className={classNames(styles['coz-modal-content'], styles['coz-modal-buttons'])}>
-            { (cancelText && cancelAction) && <button className={classNames('coz-btn', 'coz-btn--' + cancelType)} onClick={cancelAction}>
-              {cancelText}
-            </button>
-            }
-            { (validateText && validateAction) && <button className={classNames('coz-btn', 'coz-btn--' + validateType)} onClick={validateAction}>
-              {validateText}
-            </button>
-            }
-          </div>
+        { displayPrimary &&
+          <button className={classNames('coz-btn', 'coz-btn--' + primaryType)} onClick={primaryAction}>
+            {primaryText}
+          </button>
         }
       </div>
+    )
+}
+
+const Modal = (props) => {
+  props.primaryType = props.primaryType || props.validateType
+  props.primaryText = props.primaryText || props.validateText
+  props.primaryAction = props.primaryAction || props.validateAction
+  props.secondaryType = props.secondaryType || props.cancelType
+  props.secondaryText = props.secondaryText || props.cancelText
+  props.secondaryAction = props.secondaryAction || props.cancelAction
+  return (
+    <div className={styles['coz-modal-container']}>
+      <div className={styles['coz-overlay']}>
+        <div className={styles['coz-modal']}>
+          <ModalTitle {...props} />
+          <ModalCross {...props} />
+          <ModalDescription {...props} />
+          { props.children }
+          <ModalButtons {...props} />
+        </div>
+      </div>
     </div>
-  </div>
-)
+  )
+}
+
+function deprecated(arg) {
+  console.warn(`[Modal] ${arg} is deprecated, please upgrade your Modal component.`)
+}
 
 Modal.propTypes = {
   title: React.PropTypes.string.isRequired,
   description: React.PropTypes.node,
-  cancelType: React.PropTypes.string,
-  cancelText: React.PropTypes.string,
-  cancelAction: React.PropTypes.func,
-  validateType: React.PropTypes.string,
-  validateText: React.PropTypes.string,
-  validateAction: React.PropTypes.func
+  secondaryType: React.PropTypes.string,
+  secondaryText: React.PropTypes.string,
+  secondaryAction: React.PropTypes.func,
+  primaryType: React.PropTypes.string,
+  primaryText: React.PropTypes.string,
+  primaryAction: React.PropTypes.func,
+  cancelType: deprecated('cancelType') || React.PropTypes.string,
```

Comment:
@enguerran : ok, deprecated is a function that first warn the developpeur that the result will be false soon, and later is really deprecated ?